gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -349,4 +584,52 @@ private Map<String, AttributeMetadata> remapAttributeMetadataByResultName(
             .collect(Collectors.toUnmodifiableList()),
         attributeMetadataByIdMap);
   }
+
+  private Optional<org.hypertrace.gateway.service.v1.common.Filter> buildFilter(
+      org.hypertrace.gateway.service.v1.common.Filter filter,
+      AttributeSource source,
+      Map<String, AttributeMetadata> attributeMetadataMap) {
+    if (filter.equals(Filter.getDefaultInstance())) {
```

Comment:
This doesn't look right. `filter` is of the instance `org.hypertrace.gateway.service.v1.common.Filter` and `Filter.getDefaultInstance()` is of the instance `org.hypertrace.core.query.service.api.Filter`

Would always return false